nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyRecipe.groovy

```diff
@@ -216,7 +216,7 @@ class P2ProxyRecipe
   }
 
   static TokenMatcher tokenMatcherForExtensionAndName(final String extension, final String name = '.+', final String path = '.+') {
-    new TokenMatcher("{path:.*}/{name:${name}}.{extension:${extension}}")
+    new TokenMatcher("{path:${path}}/{name:${name}}.{extension:${extension}}")
```

Comment:
Path was never being used, so we actually use it now. ITs check out with this change, and I'm fine with it sneaking through.